lambdas/build-user-identity/src/main/java/uk/gov/di/ipv/core/builduseridentity/BuildUserIdentityHandler.java

```diff
@@ -215,6 +225,80 @@ private void sendIdentityIssuedAuditEvent(
+                .filter(
+                        returnCode ->
+                                configService
+                                        .getSsmParameter(RETURN_CODES_ALWAYS_REQUIRED)
+                                        .contains(returnCode.code()))
+                .toList();
+    }
+
+    private List<AuditEventReturnCode> getAuditEventReturnCodes(
+            Map<String, ContraIndicatorConfig> configMap,
+            String vot,
+            ContraIndicators contraIndicators,
+            boolean isBreachingCiThreshold) {
+        return contraIndicators.getContraIndicatorsMap().values().stream()
+                .map(
+                        contraIndicator -> {
+                            ContraIndicatorConfig contraIndicatorConfig =
+                                    configMap.get(contraIndicator.getCode());
+                            String returnCode;
+                            if (vot.equals(VectorOfTrust.P2.toString())) {
+                                returnCode =
+                                        contraIndicatorConfig != null
+                                                ? contraIndicatorConfig.getReturnCode()
+                                                : configService.getSsmParameter(
+                                                        RETURN_CODES_ALWAYS_REQUIRED);
+                            } else {
+                                returnCode =
+                                        isBreachingCiThreshold
+                                                        && contraIndicator.getCode() != null
+                                                        && contraIndicatorConfig != null
+                                                ? contraIndicatorConfig.getReturnCode()
+                                                : configService.getSsmParameter(
+                                                        RETURN_CODES_NON_CI_BREACHING_P0);
+                            }
+
+                            List<String> issuers =
+                                    contraIndicator.getIssuers() != null
+                                                    && contraIndicators.hasMitigations()
+                                            ? contraIndicator.getIssuers()
+                                            : Collections.emptyList();
+                            return Map.entry(
+                                    Objects.requireNonNull(returnCode),
+                                    issuers.stream().distinct().collect(Collectors.toList()));
+                        })
+                .collect(
+                        Collectors.toMap(
+                                Map.Entry::getKey,
+                                Map.Entry::getValue,
+                                (existing, replacement) -> {
+                                    existing.addAll(replacement);
+                                    return existing;
+                                }))
```

Comment:
This this would be more readable if we defined a variable before it and then called this variable "merged...".